fft/src/KokkosFFT_ROCM_types.hpp

```diff
@@ -17,27 +23,67 @@ static_assert(sizeof(std::complex<double>) == sizeof(Kokkos::complex<double>));
 static_assert(alignof(std::complex<double>) <=
               alignof(Kokkos::complex<double>));
 
-#ifdef ENABLE_HOST_AND_DEVICE
-#include <fftw3.h>
-#include "KokkosFFT_utils.hpp"
-static_assert(sizeof(fftwf_complex) == sizeof(Kokkos::complex<float>));
-static_assert(alignof(fftwf_complex) <= alignof(Kokkos::complex<float>));
-
-static_assert(sizeof(fftw_complex) == sizeof(Kokkos::complex<double>));
-static_assert(alignof(fftw_complex) <= alignof(Kokkos::complex<double>));
-#endif
-
 namespace KokkosFFT {
 namespace Impl {
 using FFTDirectionType                     = int;
 constexpr FFTDirectionType ROCFFT_FORWARD  = 1;
 constexpr FFTDirectionType ROCFFT_BACKWARD = -1;
 
+#if !defined(ENABLE_HOST_AND_DEVICE)
 enum class FFTWTransformType { R2C, D2Z, C2R, Z2D, C2C, Z2Z };
+#endif
 
 template <typename ExecutionSpace>
 using TransformType = FFTWTransformType;
 
+/// \brief A class that wraps rocfft for RAII
+template <typename ExecutionSpace, typename T>
+struct ScopedRocfftPlan {
+ private:
+  using floating_point_type = KokkosFFT::Impl::base_floating_point_type<T>;
+  rocfft_plan m_plan;
+  rocfft_execution_info m_execution_info;
+
+  using BufferViewType =
+      Kokkos::View<Kokkos::complex<floating_point_type> *, ExecutionSpace>;
+
+  bool m_is_plan_created = false;
+  bool m_is_info_created = false;
+
+  //! Internal work buffer
+  BufferViewType m_buffer;
+
+ public:
+  ScopedRocfftPlan() {}
```

Comment:
```suggestion
  ScopedRocfftPlan() = default;
```

Is it required though ?

Reply:
`ScopedRocfftPlan` was refactored and this constructor was removed